contracts/utils/ICS20Lib.sol

```diff
@@ -153,42 +144,40 @@ library ICS20Lib {
     }
 
     /// @notice getDenomPrefix returns an ibc path prefix
-    /// @param port Port
-    /// @param client client
+    /// @param portId Port
+    /// @param clientId client
     /// @return Denom prefix
-    function getDenomPrefix(string calldata port, string calldata client) internal pure returns (bytes memory) {
-        return abi.encodePacked(port, "/", client, "/");
+    function getDenomPrefix(string calldata portId, string calldata clientId) internal pure returns (bytes memory) {
+        return abi.encodePacked(portId, "/", clientId, "/");
+    }
+
+    /// @notice removeHop removes the first hop from the denom trace.
+    /// @param denom Denom to remove the hop from
+    /// @param hop Hop to remove (it must be the first hop)
+    /// @return The new denom with the first hop removed
+    function removeHop(bytes memory denom, bytes memory hop) internal pure returns (bytes memory) {
+        return Bytes.slice(denom, hop.length);
     }
 
-    /// @notice toIBCDenom converts a full denom path to an ibc/hash(trace+base_denom) denom
-    /// @notice there is no check if the denom passed in is a base denom (if it has no trace), so it is assumed
-    /// @notice that the denom passed in is a full denom path with trace and base denom
-    /// @param fullDenomPath full denom path with trace and base denom
-    /// @return IBC denom in the format ibc/hash(trace+base_denom)
-    function toIBCDenom(string memory fullDenomPath) public pure returns (string memory) {
-        string memory hash = toHexHash(fullDenomPath);
-        return string(abi.encodePacked(IBC_DENOM_PREFIX, hash));
+    /// @notice addHop adds a hop to the denom trace as the first hop.
+    /// @param denom Denom to add the hop to
+    /// @param hop Hop to add
+    /// @return The new denom with the hop added
+    function addHop(bytes memory denom, bytes memory hop) internal pure returns (bytes memory) {
+        return abi.encodePacked(hop, denom);
     }
```

Comment:
ditto about efficiency and the need for this function. The internal logic is easy to read.

Reply:
I removed them, but I think maybe the compiler inlines them anyway, because there was 0 gas difference in any tests.